### ISSUE TITLE
Intel preprocessor for intel builds

### ISSUE THREAD

**Issue by jacobwilliams** [jacobwilliams/json-fortran#205]
I don't know if this is possible (maybe a question for @cmacmackin), but we should use the intel preprocessor (fpp) when we build the library with ifort. So, if you call `build.sh --compiler Intel`, this should also be passed into FORD somehow to indicate to use the intel `fpp` rather than gfortran `cpp`.

**Comment by zbeekman** [comment]
Yes, I think the newest releases of FORD support this, however the other issues may need to be resolved first.

**Comment by jacobwilliams** [comment]
As suggested [here](https://github.com/cmacmackin/ford/issues/139), could do something like this in `build.sh` (would have to add `preprocessor` line to project file):

depending on compiler:
````bash
FPP='gfortran -E'
FPP='fpp'
etc.
````
Then:
````bash
sed -i.backup "s/^preprocessor:.*$/preprocessor: $FPP/" json-fortran.md
````

**Comment by zbeekman** [comment]
I think using markdown include is a better solution, due to the CMake implementation... at least until @cmacmackin adds a command line flag for preprocessor (or one of us adds one and creates a PR) I'll add this into my open PR... then after it's merged can we create a 5.0.1 release? JSON-Fortran is effectively blocking FORD's Homebrew formula from being updated due to this issue. (This is because cpp resolves to clang in the homebrew build environment....)
